src/test/regress/sql/multi_behavioral_analytics_basics.sql

```diff
@@ -476,3 +476,87 @@ FROM
   ORDER BY 1, 2;
 
 SELECT count(*), count(DISTINCT user_id), avg(user_id), avg(value_1_agg) FROM agg_results;
+
+-- test order by when pushing down avg
+SELECT user_id, avg(value_1)
+FROM users_table
+GROUP BY user_id
+ORDER BY avg(value_1) DESC
+LIMIT 5;
+
+SELECT user_id, avg(value_1)
+FROM users_table
+GROUP BY user_id
+ORDER BY avg(value_1) DESC
+LIMIT 1;
+
+SELECT user_id, avg(value_1) + 1
```

Comment:
Maybe some tests with (i) target list doesn't contain the aggregate (ii) target list and order by contains different aggregates  (iii) Same aggragate with different expression `SELECT user_id, avg(value_1) FROM users_table GROUP BY user_id ORDER BY avg(value_1) * user_id DESC LIMIT 5;`

Reply:
added some, I started to think we should also add explain tests